packages/core/src/routes/experience/verification-routes/new-password-identity-verification.ts

```diff
@@ -18,8 +19,12 @@ export default function newPasswordIdentityVerificationRoutes<
     `${experienceRoutes.verification}/new-password-identity`,
     koaGuard({
       body: z.object({
-        identifier: interactionIdentifierGuard,
-        password: z.string().optional(),
+        identifier: z.object({
+          // Only username is supported for now
+          type: z.literal(SignInIdentifier.Username),
+          value: z.string().regex(usernameRegEx),
```

Comment:
Use the profile-fulfilling endpoint for email and phone registration flow instead. 
NewPasswordIdentify only supports username + password for now.